extensions/jackson/runtime/src/main/java/io/quarkus/jackson/runtime/ObjectMapperProducer.java

```diff
@@ -63,6 +64,9 @@ public ObjectMapper objectMapper(@All List<ObjectMapperCustomizer> customizers,
         for (ObjectMapperCustomizer customizer : sortedCustomizers) {
             customizer.customize(objectMapper);
         }
+        if (true) {
```

Comment:
By default and mostly for backward compatibility reasons Jackson uses the `ThreadLocal` based pool that however isn't virtual thread friendly.

We should decide under which condition we may want to replace that default pool with something not relying on `ThreadLocal`. Personally I wouldn't leave this decision to end users, but I'd take it automatically. For instance we could use a virtual threads friendly pool when the project is running on Java 21+ and uses the `@RunOnVirtualThread` annotation at least once.